contracts/debt/IncurDebtV1.sol

```diff
@@ -0,0 +1,332 @@
+     * @param _tokenToReceiveExcess amount of OHM to borrow
+     */
+    function repayDebtWithCollateralAndWithdrawTheRest(address _tokenToReceiveExcess) external {
+        uint256 depositedCollateralAfterRepay = _repay(msg.sender);
+        assignBorrowerInfoToZero(msg.sender);
+
+        if (depositedCollateralAfterRepay > 0) {
+            if (_tokenToReceiveExcess == sOHM) {
+                IERC20(sOHM).safeTransfer(msg.sender, depositedCollateralAfterRepay);
+            } else if (_tokenToReceiveExcess == gOHM) {
+                IERC20(sOHM).approve(staking, depositedCollateralAfterRepay);
+                IStaking(staking).wrap(msg.sender, depositedCollateralAfterRepay);
+            }
+        }
+    }
+
+    /**
+     * @notice deposits OHM to pay debt
+     * - msg.sender must be a borrower
+     * - msg.sender's OHM allowance must be >= _ohmAmount
+     * - borrower must have outstanding debt
+     * @param _ohmAmount amount of OHM to borrow
+     */
+    function repayDebtWithOHM(uint256 _ohmAmount) external isBorrower(msg.sender) {
+        if (borrowers[msg.sender].debt == 0) revert IncurDebtV1_BorrowerHasNoOutstandingDebet(msg.sender);
+        IERC20(OHM).safeTransferFrom(msg.sender, address(this), _ohmAmount);
+
+        totalOutstandingDebt -= _ohmAmount;
+        borrowers[msg.sender].debt = uint128(borrowers[msg.sender].debt - _ohmAmount);
+
+        IERC20(OHM).approve(treasury, _ohmAmount);
+        ITreasury(treasury).repayDebtWithOHM(_ohmAmount);
+    }
+
+    /**
+     * @notice repays debt using collateral and returns remaining tokens to borrower
+     * - onlyOwner (or governance)
+     * - sends remaining tokens to owner in sOHM
+     * @param _borrower the address that will interact with contract
+     * @param _to where to send remaining sOHM
+     */
+    function forceRepay(address _borrower, address _to) external onlyGovernor {
+        uint256 collateralAfterDebtPayment = _repay(_borrower);
+
+        assignBorrowerInfoToZero(_borrower);
+        if (collateralAfterDebtPayment > 0) IERC20(sOHM).safeTransfer(_to, collateralAfterDebtPayment);
+    }
+
+    /**
+     * @notice seize and burn _borrowers collateral and forgive debt
+     * - will burn all collateral, including excess of debt
+     * - onlyGovernance
+     * @param _borrower the account to seize
+     */
+    function seize(address _borrower) external onlyGovernor isBorrower(_borrower) {
+        _repayDebtWithCollateralByGovernor(_borrower);
+        uint256 seizedCollateral = borrowers[_borrower].collateralInSOHM;
+
+        assignBorrowerInfoToZero(_borrower);
+        IERC20(sOHM).approve(staking, seizedCollateral);
+
+        IStaking(staking).unstake(address(this), seizedCollateral, false, true);
+        IOHM(OHM).burn(seizedCollateral);
+    }
+
+    /**
+     * @notice updates borrowers sOHM collateral to current index
+     * @param _borrower borrowers address
+     */
+    function updateCollateralInSOHM(address _borrower) public {
```

Comment:
why not just store the users collateral in gOhm instead of sOHM to avoid having to do this?

Reply:
That will be more convenient but the guide from the interface tp provide wanted this approach.

Reply:
@tonypiano any specific reason we want to hold sOHM instead of gOHM seems like gOHM is the more popular token now and also easier to work with in this contract.

Reply:
Yeah, just remembered, so the `updateCollateralInSOHM` helps the borrower increase the loan he can take so regardless if the contract takes sOHM/gOHM.

If I deposit 10 gOHM I can borrow 15 OHM after 8 hours I'll be able to borrow say 17 OHM due to rebase.

Reply:
we can also do that with gOHM. getAvailableToBorrow() uses gOHM so that value will keep increasing with rebases.

Reply:
Yeah, `updateCollateralInSOHM` automatically increases borrowers' loans can also be done manually by the borrower. ` getAvailableToBorrow()` is just a view function.

Reply:
yea but the amount they can borrow is based on getAvailableToBorrow(). So even if its a view function because its increasing it will increase the amount they can borrow over time.